Sources/Xcodeproj/XcodeProjectModelSerialization.swift

```diff
@@ -431,7 +431,11 @@ fileprivate func combineBuildSettingsPropertyLists(
         // FIXME: We should resolve `$(inherited)` here.  The way this works is
         // that occurrences of `$(inherited)` in the overlay are replaced with
         // the overlaid value in the base.
-        resultDict[name] = value
+        if let prevValue = resultDict[name] {
```

Comment:
I think this shouldn't be needed if we use `$(inherited)` in the target specific build settings. See comment below.

Reply:
The FIXME's says `$(inherited)` doesn't work. This change makes a step in that direction though so the values are combined rather than replaced.

Reply:
I am not sure if we should be resolving build settings that come from project settings. But I've not really worked in this area much. @abertelrud should be able to explain!

Reply:
I think, for now, lets leave this area unchanged as using `$(inherited)` seems like the right solution.

Reply:
maybe. The story behind this change is here: https://github.com/apple/swift-package-manager/commit/9082167e629829ffe090d6df8be73df330826326

although we "combine" common and debug `BuildSettings`, it's not really combined, it's replaced. But seems everything is empty at this stage so it doesn't really matter.

I'll revert this.